lmcache/storage_backend/local_backend.py

```diff
@@ -15,6 +18,9 @@
 
 logger = init_logger(__name__)
 
+class LocalBackendEndSignal:
+    pass
+
 class LMCLocalBackend(LMCBackendInterface):
```

Comment:
Seems like LMLocalBackend and LMRemoteBackend share a lot of codes. Maybe we should consider merging it back. 
This PR doesn't need to solve this, let's create a new issue after this PR is merged.

In my opinion, there should be 2 different scenarios:
- KV cache is in fast memory and serialization overhead is 0 or negligible --> don't need async put
- KV cache is in slower storages or have high serialization overhead --> need async put, and need pipelined get if storage is slow IO and deserialization is slow computation